homeassistant/components/sensor/hive.py

```diff
@@ -0,0 +1,486 @@
+        if snan is not None:
+            if 'now' in snan:
+                if ('value' in snan["now"] and
+                        'start' in snan["now"] and
+                        'Start_DateTime' in snan["now"] and
+                        'End_DateTime' in snan["now"] and
+                        'status' in snan["now"]["value"]):
+                    now_status = snan["now"]["value"]["status"]
+                    now_start = snan["now"]["Start_DateTime"].strftime("%H:%M")
+                    now_end = snan["now"]["End_DateTime"].strftime("%H:%M")
+
+                    sa_string = (now_status
+                                 + " : "
+                                 + now_start
+                                 + " - "
+                                 + now_end)
+                    s_a.update({"Now": sa_string})
+
+            if 'next' in snan:
+                if ('value' in snan["next"] and
+                        'start' in snan["next"] and
+                        'Start_DateTime' in snan["next"] and
+                        'End_DateTime' in snan["next"] and
+                        'status' in snan["next"]["value"]):
+                    next_status = snan["next"]["value"]["status"]
+                    nxtstrt = snan["next"]["Start_DateTime"].strftime("%H:%M")
+                    next_end = snan["next"]["End_DateTime"].strftime("%H:%M")
+
+                    sa_string = (next_status
+                                 + " : "
+                                 + nxtstrt
+                                 + " - "
+                                 + next_end)
+                    s_a.update({"Next": sa_string})
+            if 'later' in snan:
+                if ('value' in snan["later"] and
+                        'start' in snan["later"] and
+                        'Start_DateTime' in snan["later"] and
+                        'End_DateTime' in snan["later"] and
+                        'status' in snan["later"]["value"]):
+                    later_status = snan["later"]["value"]["status"]
+                    later_start = (snan["later"]
+                                   ["Start_DateTime"].strftime("%H:%M"))
+                    later_end = snan["later"]["End_DateTime"].strftime("%H:%M")
+
+                    sa_string = (later_status
+                                 + " : "
+                                 + later_start
+                                 + " - "
+                                 + later_end)
+                    s_a.update({"Later": sa_string})
+        else:
+            s_a.update({"Schedule not active": ""})
+
+        return s_a
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        if self.device_type == "Heating_CurrentTemperature":
+            return TEMP_CELSIUS
+        elif self.device_type == "Heating_TargetTemperature":
+            return TEMP_CELSIUS
+        elif self.device_type == "Heating_State":
+            return None

```

Comment:
All the cases that return `None` can be the `else` case at the end.

Reply:
done